src/runtime/Types/MetaType.cs

```diff
@@ -130,10 +143,12 @@ public static NewReference tp_new(BorrowedReference tp, BorrowedReference args,
                 using var clsDict = new PyDict(dict);
                 if (clsDict.HasKey("__assembly__") || clsDict.HasKey("__namespace__"))
                 {
-                    return TypeManager.CreateSubType(name, base_type, clsDict);
+                    return TypeManager.CreateSubType(name, baseTypes[0], interfaces, clsDict);
                 }
             }
 
+            var base_type = Runtime.PyTuple_GetItem(bases, 0);
```

Comment:
`bases` here can be empty since you removed check above.

Reply:
How can bases be length 0? This should only be called when a C# type is subclassed, right? In this case, it should at least be length 1.